sdk/identity/Azure.Identity/src/VisualStudioCodeCredential.cs

```diff
@@ -61,6 +61,11 @@ private async ValueTask<AccessToken> GetTokenImplAsync(TokenRequestContext reque
 
             try
             {
+                if (string.Equals(_tenantId, Constants.AdfsTenantId, StringComparison.Ordinal))
+                {
+                    throw new CredentialUnavailableException("VisualStudioCodeCredential authentication unavailable. ADFS tenant / authorities are not supported.");
+                }
+
                 GetUserSettings(out var tenant, out var environmentName);
```

Comment:
Although authentication for adfs tenants won't work from the IDE.
But the user input can still configure it VS CODE IDE settings and the credential here is reading it.
So, to avoid breaking the flow, we can add the check here too.

Reply:
updated this check to come after we've resolved the tenant from the config is it's available.